src/seedu/addressbook/data/person/Address.java

```diff
@@ -38,7 +64,11 @@ public static boolean isValidAddress(String test) {
 
     @Override
     public String toString() {
-        return value;
+        return String.format("%s, %s, %s, %s",
+                block.toString(),
+                street.toString(),
+                unit.toString(),
+                postalCode.toString());
     }
 
     @Override
```

Comment:
Now that you decomposed address into individual components, is there a need to update this method? 
Can you think of 2 addresses that have equal individual components yet this method return false?